### ISSUE TITLE
Specifying Ingress Rules in Spec for nuclio.deploy

### ISSUE THREAD

**Issue by RAbraham** [nuclio/nuclio-jupyter#49]
Hi,
How do I specify Ingress rules in Python? I tried the following but please correct me.

```python
spec_data = {
        "spec.triggers.http-trial1.class": "",
        "spec.triggers.http-trial1.kind": "http",
        "spec.triggers.http-trial1.maxWorkers": 2,
        "spec.triggers.http-trial1.workerAvailabilityTimeoutMilliseconds": 30,
        "spec.triggers.http-trial1.attributes.ingresses.0.paths": ['/python-trial-odin']
    }
import nuclio
spec = nuclio.ConfigSpec(config=spec_data)
nuclio.deploy_file(   "https://raw.githubusercontent.com/nuclio/nuclio/master/hack/examples/python/helloworld/helloworld.py", dashboard_url=dashboard_url, name=func_name, project=project_name, output_dir='/Users/rabraham/output',spec=spec)

```

The error I get is:
```bash
Traceback (most recent call last):
  File "/Users/rabraham/odin/nuclio_server.py", line 155, in <module>
    deploy_file_py(dashboard_url, func_name, project_name, y_d)
  File "/Users/rabraham/odin/nuclio_server.py", line 95, in deploy_file_py
    spec=spec)
  File "/Users/rabraham/odin/venv/lib/python3.6/site-packages/nuclio/deploy.py", line 82, in deploy_file
    tag=tag, verbose=verbose, create_new=create_project)
  File "/Users/rabraham/odin/venv/lib/python3.6/site-packages/nuclio/deploy.py", line 200, in deploy_config
    state, address = deploy_progress(api_address, name, verbose)
  File "/Users/rabraham/odin/venv/lib/python3.6/site-packages/nuclio/deploy.py", line 247, in deploy_progress
    ip = get_address(api_address)
  File "/Users/rabraham/odin/venv/lib/python3.6/site-packages/nuclio/deploy.py", line 259, in get_address
    raise OSError('nuclio API call failed')
OSError: nuclio API call failed

Process finished with exit code 1

```

**Comment by yaronha** [comment]
@RAbraham i'm looking into it, may relate to working w local docker, get the URL manually (`localhost:8070/api/external_ip_addresses`) see if it returns value ?

I assume u set the values for: dashboard_url=dashboard_url, name=func_name, project=project_name?

**Comment by RAbraham** [comment]
Thanks @yaronha . I should have pasted the output of `localhost:8070/api/external_ip_addresses`. The reason I was hesitant to do it because the docs said

>The user must know through which URL a function can be invoked in case load balancing / ingresses are not used. If the user concatenates one of the external IP addresses returned by this endpoint with the function's HTTP port, as specified in its spec/status - this will form a valid invocation URL.

So, I thought that I need an ingress to make it work. Anyways, the error is below.


>curl localhost:8070/api/external_ip_addresses
>{"error":"\nError - nodes is forbidden: User \"system:serviceaccount:nuclio:nuclio\" cannot list resource \"nodes\" in API group \"\" at the cluster scope\n    .../nuclio/nuclio/pkg/platform/kube/platform.go:282\n\nCall stack:\nFailed to get nodes\n    .../nuclio/nuclio/pkg/platform/kube/platform.go:282\nFailed to get nodes\n    .../nuclio/nuclio/pkg/platform/kube/platform.go:586\nFailed to get external IP addresses\n    .../dashboard/resource/externalipaddresses.go:35\n"}%


Yes, the code above did not get wrapped but I did do `dashboard_url=dashboard_url, name=func_name, project=project_name?`

**Comment by RAbraham** [comment]
This is fixed by ee59b643a3448bd013ee3619cdd98de42d48efbe. Thanks.
